rstudio/rstudio
# support R 4.2.0 ResetConsole hook on Windows

### Intent

Addresses https://github.com/rstudio/rstudio/issues/10819.

### Approach

It took some time for R Core to decide on the interface they wanted to provide for the extension of R callbacks on Windows; this PR accommodates the final interface. See https://github.com/wch/r-source/commit/42f496e2592d1ee772fd8599c69752150ff90e0b for details.

### Automated Tests

Not included.

### QA Notes

Download the latest versions of [R-devel](https://cran.r-project.org/bin/windows/base/rdevel.html). Then, try to run the following code by selecting it all and running it all at once.

```
x <- 1
stop("oh no")
x <- 2
```

Only the first two lines of code should be executed.

### Checklist
